dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/programrule/RuleActionMapper.java

```diff
@@ -0,0 +1,166 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.programrule;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import lombok.RequiredArgsConstructor;
+
+import org.hisp.dhis.rules.models.RuleAction;
+import org.hisp.dhis.rules.models.RuleActionAssign;
+import org.hisp.dhis.rules.models.RuleActionError;
+import org.hisp.dhis.rules.models.RuleActionErrorOnCompletion;
+import org.hisp.dhis.rules.models.RuleActionSetMandatoryField;
+import org.hisp.dhis.rules.models.RuleActionShowError;
+import org.hisp.dhis.rules.models.RuleActionShowWarning;
+import org.hisp.dhis.rules.models.RuleActionWarningOnCompletion;
+import org.hisp.dhis.rules.models.RuleEffects;
+import org.hisp.dhis.setting.SystemSettingManager;
+import org.hisp.dhis.tracker.bundle.TrackerBundle;
+import org.hisp.dhis.tracker.domain.Attribute;
+import org.hisp.dhis.tracker.domain.Enrollment;
+import org.hisp.dhis.tracker.domain.TrackedEntity;
+import org.hisp.dhis.tracker.domain.TrackerDto;
+import org.hisp.dhis.tracker.programrule.implementers.enrollment.AssignValueExecutor;
+import org.hisp.dhis.tracker.programrule.implementers.enrollment.ErrorWarningRuleAction;
+import org.hisp.dhis.tracker.programrule.implementers.enrollment.RuleActionExecutor;
+import org.hisp.dhis.tracker.programrule.implementers.enrollment.RuleEngineErrorExecutor;
+import org.hisp.dhis.tracker.programrule.implementers.enrollment.SetMandatoryFieldExecutor;
+import org.hisp.dhis.tracker.programrule.implementers.enrollment.ShowErrorExecutor;
+import org.hisp.dhis.tracker.programrule.implementers.enrollment.ShowErrorOnCompleteExecutor;
+import org.hisp.dhis.tracker.programrule.implementers.enrollment.ShowWarningExecutor;
+import org.hisp.dhis.tracker.programrule.implementers.enrollment.ShowWarningOnCompleteExecutor;
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.Lists;
+
+@Service
+@RequiredArgsConstructor
+class RuleActionMapper
+{
+    private final SystemSettingManager systemSettingManager;
+
+    public Map<TrackerDto, List<RuleActionExecutor>> mapEnrollmentRuleActions( List<RuleEffects> ruleEffects,
```

Comment:
At least in my mind we are mapping `RuleEffects` to `RuleActionExecutor` 😅 So for me it would either be `mapRuleEffects` or `mapToRuleActionExecutor`